### ISSUE TITLE
FEATURE: MVP (placeholder) for Course Details page

### ISSUE THREAD

**Issue by pconrad** [ucsb-cs156/proj-courses#33]
# User Story

* As a user
* I can click a button on any course to see a full page of details about that course
* So that I can better plan my schedule

# This is an Epic, not a single issue

This is an Epic, meaning that this issue is intended to be the "master issue" that is broken into multiple individual issues.

Copy the user stories/acceptance criteria into a separate  issue before starting to work on it, and make PRs for each part separately.  Benefits to this staged approach:
* You'll earn more points that way
* You'll earn them more quickly
* PRs are easier to code review and merge
* If you don't finish, you get partial credit for what you did complete, and students in a future class can finish off your work.

# Discussion

There is a lot of information about a course that we have available; too much to put in a single row.

What would be helpful is if each course had a button where you could get more course details.

There are many thing that could go on a course details page. Note that in the first iteration of this story, the course detail page might simply restate the information already available in the current app; the details page would be a place that we could then expand on that. 

Candidates for a course details page (each of these could be a separate issue / PR or even multiple ones):

* Fields that are not shown on the regular course section table, such as the course description, the GE areas that the course fulfills.
* Course descriptions. This can replace the existing "course descriptions" search page, which is currently out of place among the "Other searches" pages in the navbar.
* Information on the final exam, which we can get from a separate endpoint (but only for one course at a time) See #32 for implementation details.
* Information on grade distributions for the course where available (see: #29  for information on where this might come from)
* A brief history of the last few times this course was offered over the last n quarters (where n is a constant that is defined in only ONE place in the code, so that eventually it might be replaced by a value we set using an environment variable.)

These additional issue may involve additional backend endpoints; please consult with the staff before creating these since we want to try to coordinate the naming and formatting of those for consistency.

# MVP Issues

# F1: Frontend placeholder for course details page

- [ ] There is a page component for CourseDetailsPage which can be a placeholder for now.
- [ ] There is a storybook entry for that component
- [ ] There is a route in App.js that routes a frontend route `/coursedetails/{qyy}/{enrollCd}` to this page.
- [ ] The page component uses useParams to get the values of the {qyy} (the quarter in qyy format, e.g. F23, and the enrollCd for the course, and displays those on the placeholder page.

# F2: Button to route to placeholder page

- [ ] There is a button, hopefully small, perhaps with the logo for "more information" (ⓘ or one of the alternatives explained [here](https://stackoverflow.com/a/69679332/13960329)) beside every course in the courses table (i.e. search results) both lectures and sections that takes you to the course details page with the correct quarter and enrollcode.

### Implementation Todos

Frontend:
- [ ] There are new form / card and/or table components for displaying course information (along with tests/storybook entries)
- [ ] There is a placeholder page for the course details page with, at a minimum, quarter and enroll code (that's enough to uniquely identify the course).  It would also be good in a minimum viable product to present at least what's in the current views, so perhaps just reusing the existing table components with just one row.
- [ ] There is a route in App.js to lead to this new page, using quarter (in yyyyq or qyy format, as you see fit, and enroll code as the ApiParms, e.g. something like `/course/details/20231/08227` for CMPSC 156 in F23, or `/course/details/F23/08227` if you prefer.  (Note that yyyyq is easier for the programmer, but qyy is probably nicer for the end user.)
- [ ] There is a button in the user interface for the main course search view that takes you to this new details page for the course.
